crates/subspace-networking/src/connected_peers.rs

```diff
@@ -186,6 +189,11 @@ impl<Instance> Behaviour<Instance> {
             default_keep_alive_until
         };
 
+        // Disable connection handler.
+        if !self.config.enabled {
+            keep_alive = KeepAlive::No;
+        }
```

Comment:
Hm, this should be moved up, there is no need to touch `self.known_peers` or matching `state.connection_state` if `self.config.enables == false`, right?

I guess just `if !self.config.enabled {return Handler::new(KeepAlive::no)}` would be sufficient.